server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2076,7 +2080,7 @@ private static boolean isWritingToRepository(SnapshotsInProgress.Entry entry) {
             // Entry is writing to the repo because it's finalizing on master
             return true;
         }
-        for (ObjectCursor<ShardSnapshotStatus> value : entry.shards().values()) {
```

Comment:
This completely failed to account for clones, so deletes incorrectly assumed that a clone wasn't writing and started concurrently.

Reply:
I've come across it multiple times and did not notice it... thanks.